src/main/java/it/gov/pagopa/pu/fileshare/util/AESUtils.java

```diff
@@ -99,14 +100,21 @@ public static InputStream encrypt(String password, InputStream plainStream) {
       new CipherInputStream(new BufferedInputStream(plainStream), cipher));
   }
 
-  /** It will read and store the provided inputStream as a ciphered file using AES GCM mode configured with the provided password.<BR />
-   * If the ciphered file already exists, it will throw {@link java.nio.file.FileAlreadyExistsException} */
-  public static void encryptAndSave(String password, InputStream plainStream, Path targetPath, String fileName)
-    throws IOException {
+  /**
+   * It will read and store the provided inputStream as a ciphered file using
+   * AES GCM mode configured with the provided password.<BR /> If the ciphered
+   * file already exists, it will throw
+   * {@link java.nio.file.FileAlreadyExistsException}
+   *
+   * @return file digest
+   */
+  public static byte[] encryptAndSave(String password, InputStream plainStream, Path targetPath, String fileName)
+    throws IOException, NoSuchAlgorithmException {
     Path targetCipherFile = targetPath.resolve(fileName + CIPHER_EXTENSION);
     try (InputStream is = plainStream;
          InputStream cipherIs = encrypt(password, is)) {
       Files.copy(cipherIs, targetCipherFile);
+      return FileUtils.calculateHash(is);
```

Comment:
this way the calculateHash will have a stream already closed.
I expect something like:
```suggestion
    MessageDigest digest = MessageDigest.getInstance("SHA-256");
    try (DigestInputStream is = new DigestInputStream(plainStream, digest);
         InputStream cipherIs = encrypt(password, is)) {
      Files.copy(cipherIs, targetCipherFile);
      digest.digest()
```